arangod/Cluster/ServerState.h

```diff
@@ -57,6 +57,13 @@ class ServerState {
     STATE_SHUTDOWN        // used by all roles
   };
 
+  enum ReadOnlyMode {
```

Comment:
I would prefer if we could turn this into an `enum class`. That way we will not have any implicit conversions and only the defined enum values can be used.